File: backend/Pet-Society/src/main/java/Pet/Society/models/dto/appointment/AvailableAppointmentDTO.java
package Pet.Society.models.dto.appointment;

import Pet.Society.models.enums.Reason;
import Pet.Society.models.enums.Speciality;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@SuperBuilder
public class AvailableAppointmentDTO {
    private Long appointmentId;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private String doctorName;
    private Long doctorId;
    private Speciality doctorSpeciality;
    private Reason reason;
}

File: frontend/src/app/models/dto/appointment/available-appointment-dto.ts
import { Reason } from '../../enums/reason.enum';
import { Speciality } from '../../enums/speciality.enum';

export interface AvailableAppointmentDTO {
  appointmentId: number;
  startTime: string; // ISO string (LocalDateTime)
  endTime: string; // ISO string (LocalDateTime)
  doctorName: string;
  doctorId: number;
  doctorSpeciality: Speciality;
  reason: Reason;
}

